modules/dicom_archive/js/dicom_archive.js

```diff
@@ -1,85 +1,240 @@
-                    Type: 'Text',
-                    FormName: 'Location',
-                    Value: this.getFormValue("Location")
-                }),
-                React.createElement(FilterActions, { Module: 'dicom_archive' })
-            )
-        );
+/* exported RDicomArchive */
+/* global QueryString  */
+
+/**
+ * DICOM Archive Page.
+ *
+ * Renders DICOM Archive main page consisting of FilterTable and
+ * DataTable components.
+ *
+ * @author Alex Ilea
+ * @version 1.0.0
+ *
+ * */
+var DicomArchive = React.createClass({
+  displayName: 'DicomArchive',
+
+  propTypes: {
+    Module: React.PropTypes.string.isRequired
+  },
+  mixins: [React.addons.PureRenderMixin],
+  getInitialState: function getInitialState() {
+    return {
+      Filter: {}
+    };
+  },
+  getDefaultProps: function getDefaultProps() {
+    return {
+      Gender: {
+        M: 'Male',
+        F: 'Female',
+        O: 'N/A'
+      }
+    };
+  },
+  componentDidMount: function componentDidMount() {
+    var formRefs = this.refs;
+    var queryString = new QueryString();
+    var queryStringObj = queryString.get();
+
+    // Populate input fields from query string
+    Object.keys(queryStringObj).map(function (key) {
+      if (formRefs[key].state && queryStringObj[key]) {
+        formRefs[key].state.value = queryStringObj[key];
+      }
+    });
+
+    this.setState({
+      Filter: queryStringObj,
+      QueryString: queryString
+    });
+  },
+  setFilter: function setFilter(fieldName, fieldValue) {
+    // Create deep copy of a current filter
+    var Filter = JSON.parse(JSON.stringify(this.state.Filter));
+    var queryString = this.state.QueryString;
+    var formRefs = this.refs;
+
+    // If fieldName is part of the form, add to querystring
+    if (formRefs.hasOwnProperty(fieldName)) {
+      queryString.set(Filter, fieldName, fieldValue);
+    } else {
+      queryString.clear(this.props.Module);
+    }
+
+    // Special treatment for site, to explicitly set it as an integer value
+    if (fieldName === "site") {
+      var number = Number.parseInt(fieldValue, 10);
+      if (Number.isInteger(number)) {
+        fieldValue = number;
+      }
     }
 
+    if (fieldValue === "") {
+      delete Filter[fieldName];
+    } else {
+      Filter[fieldName] = fieldValue;
+    }
+
+    this.setState({ Filter: Filter });
+  },
+  clearFilter: function clearFilter() {
+    var queryString = this.state.QueryString;
+    var formRefs = this.refs;
+
+    // Clear query string
+    queryString.clear(this.props.Module);
+
+    // Reset state of child components of FilterTable
+    Object.keys(formRefs).map(function (ref) {
+      if (formRefs[ref].state && formRefs[ref].state.value) {
+        formRefs[ref].state.value = "";
+      }
+    });
+
+    // Clear filter
+    this.setState({ Filter: {} });
+  },
+  toCamelCase: function toCamelCase(str) {
+    return str.replace(/(?:^\w|[A-Z]|\b\w|\s+)/g, function (match, index) {
+      if (+match === 0) return "";
```

Comment:
@driusan Are you referring to `toCamelCase()`? Because, I though you were referring to this snippet, which is also used often in react modules 

```js
// Reset state of child components of FilterTable
Object.keys(formRefs).map(function (ref) {
if (formRefs[ref].state && formRefs[ref].state.value) {
  formRefs[ref].state.value = "";
  }
});
```

In any case, both could potentially be outside of React. I created a `util` folder, so we can add a file with generic util functions there.